src/vellum/client/core/serialization.py

```diff
@@ -184,6 +184,10 @@ def _convert_mapping(
             converted_object[_alias_key(key, type_, direction, aliases_to_field_names)] = (
                 convert_and_respect_annotation_metadata(object_=value, annotation=type_, direction=direction)
             )
+
+    if "custom_parameters" in converted_object:
+        converted_object["custom_parameters"] = _handle_json_schema_conversion(converted_object["custom_parameters"])
```

Comment:
We have a `_compile_annotation` in functions.py that we already use to compile a pydantic model into a json schema. We should abstract that out somehow and reuse it instead.

But again, we should not edit the client directory, and this should instead move to the inline prompt node handling of prompt parameters